### PR TITLE
Updated pre-commit hooks

### PR DESCRIPTION
**Pull request by tmarshall**
This PR updates the pre-commit hooks on this repo. Most significantly it replaces [Black](https://github.com/psf/black) with [Ruff](https://github.com/astral-sh/ruff), and includes formatting changes on all files.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -2,21 +2,24 @@ repos:
...
+      - id: ruff
+        args: [--fix, --exit-non-zero-on-fix]
+      - id: ruff-format
   - repo: https://github.com/asottile/pyupgrade
-    rev: v3.2.2
+    rev: v3.3.1
     hooks:
       - id: pyupgrade
-        args: ["--py310-plus"]
+        args: ["--py38-plus", "--keep-mock"]
```
Comment (kfoxb) [inline .pre-commit-config.yaml:25]: I'm not very familiar with pyupgrade, why --py38-plus instead of --py310-plus?

Comment (tmarshall) [reply .pre-commit-config.yaml:25]: This was due to some copy paste from another repo. I will put it back to `py310`.

Comment (mtagle) [reply .pre-commit-config.yaml:25]: since I have a pr that moves our python version back to 3.8, we may want to keep this at 3.8, actually.

Comment (tmarshall) [reply .pre-commit-config.yaml:25]: set back.

**Review by kfoxb — APPROVED** [review]

**Pull request merged**